sound/soc/sof/intel/hda.c

```diff
@@ -443,7 +334,7 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	/* set DSP arch ops */
 	sdev->arch_ops = &sof_xtensa_arch_ops;
 
-	chip = get_chip_info(pci->device);
+	chip = (struct sof_intel_dsp_desc *)sdev->pdata->desc->pdata;
```

Comment:
that's just horrible. as mentioned above, it'd be less ambiguous with sdev->pdata->desc->chip_info

Reply:
also it doesn't hurt to use a helper to hide the actual representation. get_chip_info could be modified to return the right pointer to a structure.